server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -1037,6 +1042,12 @@ public synchronized void completeRelocationHandoff() {
     public synchronized void activateWithPrimaryContext(PrimaryContext primaryContext) {
         assert invariant();
         assert primaryMode == false;
+        // TODO: remove this check after backporting to 7.x
+        if (primaryContext.checkpoints.containsKey(shardAllocationId) == false) {
+            // can happen if the old primary was on an old version
+            assert indexSettings.getIndexVersionCreated().onOrAfter(Version.V_8_0_0);
```

Comment:
should this be "before Version.V_8_0_0" and not "onOrAfter"?

Reply:
sorry for this :(